src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -92,3 +96,72 @@ def spark_version_compatibility(spark_version: str | None) -> str:
     if (10, 0) <= version < (11, 3):
         return "kinda works"
     return "supported"
+
+
+def _check_spark_conf(conf: dict[str, str], source) -> list[str]:
+    failures = []
+    for k in INCOMPATIBLE_SPARK_CONFIG_KEYS:
+        if k in conf:
+            failures.append(f"unsupported config: {k}")
+    for value in conf.values():
+        if "dbfs:/mnt" in value or "/dbfs/mnt" in value:
+            failures.append(f"using DBFS mount in configuration: {value}")
+    # Checking if Azure cluster config is present in spark config
+    if _azure_sp_conf_present_check(conf):
+        failures.append(f"{_AZURE_SP_CONF_FAILURE_MSG} {source}.")
+    return failures
+
+
+def _safe_get_cluster_policy(ws: WorkspaceClient, policy_id: str) -> Policy | None:
+    try:
+        return ws.cluster_policies.get(policy_id)
+    except NotFound:
+        logger.warning(f"The cluster policy was deleted: {policy_id}")
+        return None
+
+
+def _check_cluster_policy(ws: WorkspaceClient, cluster, source):
+    failures = []
+    policy = _safe_get_cluster_policy(ws, cluster.policy_id)
+    if policy:
+        if policy.definition:
+            if _azure_sp_conf_present_check(json.loads(policy.definition)):
+                failures.append(f"{_AZURE_SP_CONF_FAILURE_MSG} {source}.")
+        if policy.policy_family_definition_overrides:
+            if _azure_sp_conf_present_check(json.loads(policy.policy_family_definition_overrides)):
+                failures.append(f"{_AZURE_SP_CONF_FAILURE_MSG} {source}.")
+    return failures
+
+
+def _check_cluster_init_script(ws: WorkspaceClient, init_scripts, source):
```

Comment:
if we're not doing mixins, then why this function is defined so far away from `_get_init_script_data`?

Reply:
move them to mixin and make these two functions next to each other.